utils/find_replace.py

```diff
@@ -0,0 +1,128 @@
+"""
+    Python script to upload and replace broken links with new ones.
+
+    Example Usage: python utils/find_replace.py --csv input.csv --out output.csv
+
+    Outputs a csv with meeting_id previous_url previous_filesystem_path new_file_url file_id event_file_id columns
+"""
+import argparse
+import csv
+import logging
+import os
+import shutil
+import tempfile
+from collections import namedtuple
+
+from pmg import app, db
+from pmg.models.resources import CommitteeMeeting, EventFile, File
+
+
+def create_logger(args):
+    logging.basicConfig(level=logging.INFO, filemode="a", encoding="utf-8")
+    formatter = logging.Formatter("%(asctime)s | %(levelname)s | %(message)s")
+    logger = logging.getLogger(__name__)
+    file_handler = logging.FileHandler(args.logfile)
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+    return logger
+
+
+def read_csv(path):
+    with open(path, "r") as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            if row["possible_duplicate"] == "True":
+                continue
+            yield row
+
+
+def find_meeting_by_id(id):
+    meeting = db.session.query(CommitteeMeeting).filter_by(id=int(id)).first()
+    return meeting
+
+
+def upload_file(path):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = os.path.join(temp_dir, os.path.basename(path))
+        shutil.copyfile(path, temp_path)
+        file = File()
+        file.from_file_blob(path, temp_path)
+        return file
+
+
+def process_and_upload_files(args):
+    logger = create_logger(args)
+    csv_row = namedtuple(
+        "Row",
+        "meeting_id previous_url previous_filesystem_path new_file_url file_id event_file_id",
+    )
+    for row in read_csv(args.csv):
+        row_url = row["url"]
+        meeting_id = row["meeting_id"]
+        filesystem_path = row["filesystem_path"]
+        meeting = find_meeting_by_id(meeting_id)
+        if not meeting:
+            logger.error("Could not find meeting {}".format(meeting_id))
+            continue
+        new_file_url = ""
+        try:
+            with app.app_context():
```

Comment:
Try moving this to main()

Reply:
Done, works just fine when the app context is initialized on main.